andela/jubilee-backend
# #167733651 Users should be able to make one-way trip request

### What does this PR do?
It allows users to be able to make a one-way trip request

### Description of Task to be completed?
- Wrote tests to test functionality
- added request controller
- added trip request middleware
- added trip input validations

### How should this be manually tested?
- Clone the repo and cd into app root
- Setup necessary environment variables for testing environment (check .sample.env)
- Run docker-compose up --build to build docker image or run locally using npm run start:dev
- Run npm run pretest to migrate and seed database
- run npm run test to run all tests
- On postman, run localhost:3000/api/trip/request with trip request json parameters to post a trip request: 

### Any background context you want to provide?
none

### screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/48371156/64803999-e8866800-d585-11e9-9aac-990e38a2fd9b.png)


### What are the relevant pivotal tracker stories?
[#167733651](https://www.pivotaltracker.com/story/show/167733651)
